init.lua

```diff
@@ -50,18 +49,18 @@ local function bind(...)
 end
 local function ScanXtar()
 	if mq.TLO.Me.XTarget() > 0 then
-		for i = 1, 20 do
+		for i = 1, mq.TLO.Me.XTargetSlots() do
 			local xName = mq.TLO.Me.XTarget(i).Name() or 'NULL'
 			local xHp = mq.TLO.Me.XTarget(i).PctHPs() or -1
 			local xId = mq.TLO.Me.XTarget(i).ID() or -1
 			local xType = mq.TLO.Me.XTarget(i).Type() or '?'
-
-			if (mq.TLO.Me.XTarget() > 0 and xId == 0) or xType == 'Corpse' then
+			local xCount = mq.TLO.Me.XTarget() or 0
+			if (xCount > 0 and xId == 0) or xType == 'Corpse' then
```

Comment:
`XTarget()` always returns a number, the `or` is never going to do anything.

You're already checking this on line 51 though, so you don't need it.